File: read_docs.py
import zipfile
import xml.etree.ElementTree as ET
import glob
import os

for path in glob.glob(r"c:\Projects\Portfolio\files\*.docx"):
    text = []
    try:
        with zipfile.ZipFile(path) as docx:
            if 'word/document.xml' in docx.namelist():
                tree = ET.parse(docx.open('word/document.xml'))
                root = tree.getroot()
                ns = {'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'}
                for para in root.iter(f'{{{ns["w"]}}}p'):
                    para_text = "".join([node.text for node in para.iter(f'{{{ns["w"]}}}t') if node.text])
                    if para_text:
                        text.append(para_text)
        print(f"\n--- {os.path.basename(path)} ---")
        print("\n".join(text[:30])) # print first 30 paragraphs
    except Exception as e:
        print(f"Error reading {os.path.basename(path)}: {e}")

File: read_docs.js
import { execSync } from 'child_process';
import fs from 'fs';
import path from 'path';

const files = [
    'files/Agile PRBL Report.docx',
    'files/Cognitive guardian.docx',
    'files/retail pos.docx'
];

files.forEach(file => {
    try {
        const tempName = `tmp_${Date.now()}_${Math.floor(Math.random()*1000)}`;
        const zipFile = tempName + '.zip';
        fs.copyFileSync(file, zipFile);
        
        execSync(`powershell -Command "Expand-Archive -Path '${zipFile}' -DestinationPath '${tempName}' -Force"`);
        const xmlPath = path.join(tempName, 'word', 'document.xml');
        const xml = fs.readFileSync(xmlPath, 'utf8');
        
        const matches = xml.match(/<w:t[^>]*>(.*?)<\/w:t>/g) || [];
        const text = matches.map(m => m.replace(/<[^>]+>/g, '')).join('');
        
        console.log(`\n=== ${file} ===\n`);
        console.log(text.substring(0, 500) + '...');
        
        execSync(`powershell -Command "Remove-Item -Recurse -Force '${tempName}'"`);
        fs.unlinkSync(zipFile);
    } catch (e) {
        console.error(`Error reading ${file}:`, e.message);
    }
});
